Modules/rat_rescue.py

```diff
@@ -95,6 +96,17 @@ def __init__(self, case_id: UUID, client: str, system: str, irc_nickname: str,
         self._lang_id = lang_id
         self._status = status
 
+        # post-init cleanup
+        self.changedAttr = []
+
+    def __setattr__(self, key, value):
+        """set an attribute"""
+        # call the super to render effect
+        super().__setattr__(key, value)
+        # and do a sanity check to prevent recursion (Bad)
+        if key not in self.changedAttr and key != 'changedAttr':
+            self.changedAttr.append(key)
```

Comment:
You'll find that this is the perfect use case for a set, as you would not have to check whether it's already in the set.

Reply:
You know what you are absolutely right. I completely overlooked that property of Sets.